ArduCopter/mode_auto.cpp

```diff
@@ -771,33 +771,99 @@ void Copter::ModeAuto::wp_run()
         set_throttle_takeoff();
         return;
     }
-
-    // process pilot's yaw input
     float target_yaw_rate = 0;
-    if (!copter.failsafe.radio) {
-        // get pilot's desired yaw rate
-        target_yaw_rate = get_pilot_desired_yaw_rate(channel_yaw->get_control_in());
-        if (!is_zero(target_yaw_rate)) {
-            auto_yaw.set_mode(AUTO_YAW_HOLD);
-        }
-    }
 
-    // set motors to full range
+    //Enable course locked add a2sandres
+    if (g.enbl_crs_lock == 1) {

```

Comment:
The code for this feature doesn't make sense to me at all. The code says that if the feature is enabled and there **is** pilot yaw input then we look at next waypoint. The description of the parameter doesn't say that though.

Reply:
The ENB_CRS_LOCK behavior is eliminated because this function is performed by the WP_YAW_BEHAVIOR parameter